lib/compress/zstd_compress.c

```diff
@@ -2451,7 +2592,27 @@ ZSTD_blockCompressor ZSTD_selectBlockCompressor(ZSTD_strategy strat, ZSTD_dictMo
     ZSTD_STATIC_ASSERT((unsigned)ZSTD_fast == 1);
 
     assert(ZSTD_cParam_withinBounds(ZSTD_c_strategy, strat));
-    selectedCompressor = blockCompressor[(int)dictMode][(int)strat];
+    DEBUGLOG(4, "Selected block compressor: dictMode=%d strat=%d rowMatchfinder=%d", (int)dictMode, (int)strat, (int)rowMatchfinderMode);
+    if (ZSTD_strategyMayUseRowBasedMatchfinder(strat) && rowMatchfinderMode != ZSTD_urm_disableRowMatchfinder) {
```

Comment:
nit: Why check against not disabled here? It is logically simpler to check if it is enabled.

Also, can you add an assert that it isn't auto mode?